libs/langchain-core/src/messages/function.ts

```diff
@@ -40,25 +52,30 @@ export class FunctionMessage extends BaseMessage {
   }
 
   _getType(): MessageType {
-    return "function";
+    return this.type;
   }
 }
 
 /**
  * Represents a chunk of a function message, which can be concatenated
  * with other function message chunks.
  */
-export class FunctionMessageChunk extends BaseMessageChunk {
+export class FunctionMessageChunk<
+  TStructure extends $MessageStructure = $StandardMessageStructure
+> extends BaseMessageChunk<TStructure, "function"> {
   static lc_name() {
     return "FunctionMessageChunk";
   }
 
+  readonly type = "function" as const;
+
   _getType(): MessageType {
```

Comment:
_nit_: now that `type` is abstract, we don't need to define `_getType` as abstract anyway